Ada learning 0003/refactoring/speadsheat+gmail notification

a-da opened the pull request.

Commit:
Check google spreadsheet and send checking summary by the email

Commit:
fixed run command

Commit:
Check google spreadsheet and send checking summary by the email: fix make run command

Commit:
Halt instead raising the error based on assertion

Commit:
ada_google_com V0.1.0::

- Store configuration as YAML instead of JSON,
main benefit is allow comments and hints inside configuration.
- Redesign sheets.py object data structure process based on pydantic data models.
- Increase test coverage from 11 to 59%.

galij899 commented on google.com/src/ada_google_com/sheets.py:
```diff
@@ -1,133 +1,239 @@
+        )
+        match = regexp.search(notify_after_n_days_raw_value)
+        if not match:
+            print(f'Can not match {regexp.pattern}')
+            sys.exit(1)
```
Why using such rough syntax? Can't a proper exception be raised there?

a-da replied:
fixed

galij899 commented on google.com/src/ada_google_com/sheets.py:
```diff
@@ -1,133 +1,239 @@
+        if (now - date) > timedelta(days=self.foot.notify_after_n_days):
+            return (
+                'You have to log into to refresh account, '
+                f'you could loose that account .... {repr(name)!r}, {user!r}, {email!r}'
```
loose -> lose

galij899 replied:
Also repr(name) is unnecessary if **!r** flag is used in f-string

a-da replied:
fixed.

Commit:
ada_google_com V0.1.0, after review::

- add spelling check with enchant
- remove double repr from the log
- properly raise exc.

a-da commented:
thanks @galij899
